document/core/exec/runtime.rst

```diff
@@ -678,51 +701,67 @@ This definition allows to index active labels surrounding a :ref:`branch <syntax
+     \CAUGHTadm\{\tagaddr~\val^\ast\}~\XT~\END \\ &&|&
      \FRAME_n\{F\}~\XT~\END \\
    \end{array}
 
-Throw contexts allow matching the program context around a throw instruction up to the nearest enclosing :math:`\CATCHN` handler, thereby selecting the exception handler responsible for an exception.
+Throw contexts allow matching the program context around a throw instruction up to the innermost enclosing |CATCHadm| or |DELEGATEadm|, thereby selecting the exception |handler| responsible for an exception, if one exists.
 
 .. note::
-   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try>` would be as follows.
+   Contrary to block contexts, throw contexts don't skip over handlers.
+
+   Since handlers are not included above, there is always a unique maximal throw context to match the reduction rules.
+   This basically means that :math:`\CAUGHTadm {\dots} instr^\ast \END` is not a potential catching block for exceptions thrown by :math:`\instr^\ast`, since these are instructions in the scope of a |CATCH| or a |CATCHALL|.
+
+.. note::
+   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try-catch>` would be as follows.
+
+   Assume that :math:`\expand_F(bt) = [t1^n] \to [t2^m]`, for some :math:`n > m` such that :math:`t1^n[(n-m):n] = t2^m`,
+   and that the tag address `a` of :math:`x` corresponds to the tag type :math:`[t2^m] \to []`.
 
    .. math::
       \begin{array}{ll}
-      & \hspace{-5ex} S;~F;~\val^n~(\TRY~\X{bt}~\THROW~x~\CATCH~\RETURN~\END) \\
-      \stepto & S;~F;~\CATCHN_1\{\RETURN\}~(\LABEL_1 \{\}~\val^n~\THROWADDR~a~\END)~\END \\
+      & \hspace{-5ex} S;~F;~\val^n~(\TRY~\X{bt}~(\THROW~x)~\CATCH~x~\RETURN~\END) \\
+      \stepto & S;~F;~\LABEL_m\{\} (\CATCHadm\{a~\RETURN\}~\val^n~(\THROW~x)~\END)~\END \\
       \end{array}
 
-   :ref:`Handling <exec-throwaddr>` the thrown exception address :math:`a` in the throw context
-   :math:`T=\LABEL_1 \{\}[\_]~\END`, with the exception handler :math:`\CATCHN_1\{\RETURN\}` gives:
+   Denote :math:`\val^n = \val^{n-m} val^m`.
+   :ref:`Handling the thrown exception <exec-throwadm>` with tag address :math:`a` in the throw context
+   :math:`T=[val^{n-m}\_]`, with the exception handler :math:`H=\CATCHadm\{a~\RETURN\}` gives:
 
    .. math::
       \begin{array}{lll}
-      \stepto & S;~F;~\LABEL_1~\{\}~(\REFEXNADDR~a~\val^n)~\RETURN~\END & \hspace{9ex}\ \\
-      \stepto & (\REFEXNADDR~a~\val^n) & \\
+      \stepto & S;~F;~\LABEL_m\{\}~(\CAUGHTadm\{a~\val^m\}~\val^m~\RETURN~\END)~\END & \hspace{9ex}\ \\
+      \stepto & \val^m & \\
       \end{array}
 
-   When a throw occurs, execution halts until that throw is the continuation of a throw context in a catching try block.
+   When a throw occurs, we pop the values :math:`val^m:[t^m]` and append them to the tag address :math:`a` into
+   a |CAUGHTadm| instruction. We then search for the maximal surrounding throw context `T`, which means we pop
```

Comment:
Which `caught` instruction do we append the values..? When a throw occurs, do we even know which `catch` will catch the exception? If we don't know even whether there's any `catch` is around to catch the exception, which `caught` are we talking about here?

Reply:
@aheejin  this is indeed unclear, I am updating it to 

> When a throw of the form :math:`val^m (throw a)` occurs, we search for the maximal surrounding throw context `T`, which means we pop any other values, labels, frames, and |CAUGHTadm| instructions surrounding the throw :math:`val^m (throw a)`, until we find an exception handler (corresponding to a try block) that :ref:`handles the exception <syntax-handler>`.
> We then append the values :math:`val^m:[t^m]` to the tag address :math:`a` into a new |CAUGHTadm| instruction which we push on the stack.

Does this make more sense now?